ports/qemu-arm/Makefile

```diff
@@ -51,16 +59,69 @@ endif
 LDFLAGS= -T $(LDSCRIPT) --gc-sections -Map=$(@:.elf=.map)
 LIBS = $(shell $(CC) $(CFLAGS) -print-libgcc-file-name)
 
-SRC_C = \
+SRC_C += \
+	mcu/arm/startup.c \
+	shared/runtime/semihosting_arm.c \
+
+endif
+
+ifeq ($(QEMU_ARCH),riscv32)
+
+CROSS_COMPILE ?= riscv64-unknown-elf-
+
+GCC_VERSION = $(word 1, $(subst ., , $(shell $(CC) -dumpversion)))
+
+RV32_ABI = ilp32
+
+QEMU_ARGS += -bios none
+
+# GCC 10 and lower do not recognise the Zicsr extension in the
+# architecture name.  "Make" unfortunately does not provide any simple way
+# to perform numeric comparisons, so to keep things simple we assume that
+# GCC is at least version 10 for the time being.
+ifeq ($(GCC_VERSION),10)
```

Comment:
Maybe you may want to replace this bit with `ifeq ($(shell test $(GCC_VERSION) -ge 10; echo $$?),0)` and remove the second sentence in the original comment.  I found a solution about that issue long after the code was committed...

Reply:
Good idea, done.  But I changed `-ge` to `-le` because I think that's what you meant.